homeassistant/components/arcticspa/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""The Arctic Spa integration."""
+from __future__ import annotations
+
+import logging
+
+from pyarcticspas import Spa
+from pyarcticspas.error import SpaHTTPException, TooManyRequestsError, UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import ArcticSpaDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up an Arctic Spa from a config entry."""
+
+    device = Spa(entry.data[CONF_API_KEY])
+
+    # The API has no login endpoint. We will use a status check to test access and availability.
+    try:
+        await device.async_status()
+    except UnauthorizedError as ex:
+        raise ConfigEntryError("Invalid API token") from ex
+    except TooManyRequestsError as ex:
+        raise ConfigEntryNotReady("API overloaded, please try later") from ex
+    except SpaHTTPException as ex:
+        raise ConfigEntryNotReady(f"API returned {ex.code} {ex.msg}") from ex
```

Comment:
I recently learnt that the coordinator raises these exception when the UpdateFailed at the first refresh, so we could remove this

Reply:
This also means that you should catch UnauthorizedError and raise ConfigEntryError in the coordinator

Reply:
great news, removed!

I added the UnauthorizedError check to coordinator's `_async_update_data`.